### ISSUE TITLE
Stop recoding runtime exception

### ISSUE THREAD

**Issue by skewyapp** [skewyapp/skewy1-0#10]
**Describe the bug**
* Runtime exception when trying to call stop record
* Mostly happens in background, however foreground crashes also present

**To Reproduce**
* Unable to reproduce so far

**Expected behavior**
* Recorder to stop and not crash / throw exception

**Screenshots**
* n/a

**Smartphone (please complete the following information):**
 - All OS / DevicesVersion
 - App release 7.0

**Additional context**
Exception java.lang.RuntimeException:
  at android.app.ActivityThread.handleServiceArgs (ActivityThread.java:4703)
  at android.app.ActivityThread.-$$Nest$mhandleServiceArgs
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2181)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:201)
  at android.os.Looper.loop (Looper.java:288)
  at android.app.ActivityThread.main (ActivityThread.java:7898)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:548)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:936)
Caused by java.lang.IllegalStateException: stop() called on an uninitialized AudioRecord.
  at android.media.AudioRecord.stop (AudioRecord.java:1392)
  at com.example.skewy.service.RecorderService.stopRecording (RecorderService.kt:190)
  at com.example.skewy.service.RecorderService.killService (RecorderService.kt:91)
  at com.example.skewy.service.RecorderService.onStartCommand (RecorderService.kt:121)
  at android.app.ActivityThread.handleServiceArgs (ActivityThread.java:4685)

**Comment by skewyapp** [comment]
Since unable to reproduce, one fix should be by adding check whether recorder is initialized

```
    fun stopRecording() {
        isRecording = false
        repository.isRecording.value = isRecording
        Log.d(TAG, "stopRecording: recorder state ${recorder.state}")
        if (recorder.state == AudioRecord.STATE_INITIALIZED) { // Initialized = 1, uninitialized = 0
            recorder.stop()
            recorder.release()
        }
    }
```

**Comment by skewyapp** [comment]
try {
            stopRecording()
        }catch (exception: Exception) {
            Log.e(TAG, "recordAudio: Error stopping recorder", exception)
        }

**Comment by skewyapp** [comment]
Fixed in 8.0 with just not swallowing the error. The error will still exist, however it was impossible to reproduce. And since it happens with "stop recording", it should not affect the user in a bad way.
